File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> book;
    
    for (auto p : participant){
        book[p]++;
    }
    
    for (auto p : completion){
        book[p]--;
    }
    
    for (auto p : book) {
        if (p.second > 0){
            answer = p.first;
            break;
        }
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.io.*;
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        for (String p : participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            map.put(c, map.get(c) - 1);
        }
        
        for (String k : map.keySet()){
            if (map.get(k) != 0) {
                answer = k;
                break;
            }
        }
        
        return answer;
    }
}